### DISCUSSION TITLE
matrix norm

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23845] [Q&A General] [answered]
Hello all，
I have two questions: 
1. Can the user provide the preprocessor pmat?
2. How to output matrix norm?
thanks

**Comment by lindsayad** [comment, marked as answer]
1. MOOSE almost always supplies the preconditioning matrix to PETSc. You control what goes into the preconditioning matrix via how `computeQpJacobian` and `computeQpOffDiagJacobian` methods are implemented and what you supply in the `Preconditioning` block of your input file
2. This is not possible to do from the command line or MOOSE input file currently. In your application though you could use the API: https://petsc.org/release/docs/manualpages/Mat/MatNorm/

**Reply by nanguaxiaofendui** [reply]
thank you
